services/worker-manager/src/data.js

```diff
@@ -0,0 +1,52 @@
+const Entity = require('azure-entities');
+
+const WorkerType = Entity.configure({
+  version: 1,
+  partitionKey: Entity.keys.StringKey('name'),
+  rowKey: Entity.keys.ConstantKey('workerType'),
+  properties: {
+    name: Entity.types.String,
+    provider: Entity.types.String,
+    description: Entity.types.String,
+    created: Entity.types.Date,
+    lastModified: Entity.types.Date,
+    config: Entity.types.JSON,
+    errors: Entity.types.JSON,
+    owner: Entity.types.String,
+    providerData: Entity.types.JSON, // providers can use this to remember values between provisioning runs
```

Comment:
Also, it might make sense to provide a more generic way of storing data for providers, since this might easily hit the 1MB limit.  Perhaps a table of ProviderData with partitionKey = (provider, workerType) and rowKey = whatever the provider wants to use, and with a single JSON `data` column.  Then providers could access the table directly and could use whatever concurrency approach they want to.

Reply:
I will circle around and move this out into its own table in the next PR I'm working on.